src/lib/validate-input.js

```diff
@@ -357,3 +359,15 @@ export function validateTranposeParams(input, {permutation}) {
   }
 }
 
+export function validateNotParams(input) {
+  for (let i = 0; i < sizeOfShape(input.shape); ++i) {
+    const a = input.getValueByIndex(i);
+    if (!Number.isInteger(a)) {
+      throw new Error('Invalid input value, it should be an integer in the interval [0, 255]');
```

Comment:
```suggestion
      throw new Error('Invalid input value - it should be an integer in the interval [0, 255]');
```

or

```diff
-      throw new Error('Invalid input value, it should be an integer in the interval [0, 255]');
+      throw new Error('Invalid input value: it should be an integer in the interval [0, 255]');
```

or

```diff
-      throw new Error('Invalid input value, it should be an integer in the interval [0, 255]');
+      throw new Error('Invalid input value. It should be an integer in the interval [0, 255]');
```

(minor grammar) An em dash or colon would be more appropriate here, as they set off extra information such as explanatory phrases or supplemental facts in a later clause that explains or expands upon on what precedes it. Commas are more for dependent clauses.

Reply:
Btw, did you intend to use a distinct error message for these two cases? Otherwise we could collapse 7 lines to 3:

```
      if (!Number.isInteger(a) || a < 0 || a > 255) {
        throw new Error('Invalid input value - it should be an integer in the interval [0, 255]');
      }
```